### ISSUE TITLE
Can no longer assign phred score higher than max score.

### ISSUE THREAD

**Issue by joshuak94** [seqan/seqan3#609]
https://github.com/seqan/seqan3/blob/master/test/snippet/alphabet/quality/phred42.cpp#L22

This previously automatically converted the score to 41, but now due to the line below, it throws an error:
https://github.com/seqan/seqan3/blob/master/include/seqan3/alphabet/detail/alphabet_base.hpp#L191

Should the snippet be changed, or the assign_rank implementation be changed?

**Comment by smehringer** [comment]
Good catch. So the documentation clearly states that higher values are casted down to 41 but as I see this was only the case for `phred42`. For `phred63` it seems that assigning a higher value asserts. @mariehoffmann you implemented the quality alphabets correct? Did you intentionally not assert for phred42? 
If so, what was the reason behind it. If not I think we should change the documentation such that every alphabet behaves the same.

**Comment by mariehoffmann** [comment]
The current behaviour is correct, see `quality/all.hpp` for an in-depth explanation and a table which states the legal input ranges, the mapping ranges, and the assert ranges. See https://en.wikipedia.org/wiki/FASTQ_format#Encoding to understand why it was not straight-forward to join all the possible input formats. phred42 is casting implicitly and only asserts above 62, but phred63 does not implicitly casts and immediately asserts for input above 63. This is intended.
In short, phred42 is the most usual format and quality scores assigned directly to read output by a sequencer do not exceed that range. However when merging reads, quality scores can be improved and may exceed the practical best score, that's why phred42 is not asserting. But you probably don't want to down-cast very high scores and rather use phred63 or should check if you haven't scores from Solexa or older Illumina standard.
So, for the generic test for quality scores, you have to write more specific once respecting the assert ranges in the table `quality/all.hpp`.

**Comment by smehringer** [comment]
> But you probably don't want to down-cast very high scores and rather use phred63 or should check if you haven't scores from Solexa or older Illumina standard.

This sounds like we actually should assert on the range phred42 [42,...,62]?

**Comment by mariehoffmann** [comment]
By merging reads you can achieve scores higher than 42 without that you want to cast to another quality type. The limit of the next larger data type is a good threshold.

**Comment by h-2** [comment]
Ah, well, there are few things here. 

1) A part of the phred-range should be down-converted to a valid phred range as @mariehoffmann pointed out.
2) before my refactoring phread alphabets could be {} initialised with rank which was broken, because actually wanted initialisation by phred; this was tolerated because rank and phred were identical for 2/3 phred types.
3) after the refactoring {} init actually does init by phred
4) assignging an out-of-range RANK should throw and the behaviour is correct. 
5) assigning an out-of-range PHRED (by assign_phred or init'ing by {}) should down-convert to the highest valid phred.
